docs/adrs/003-store-room-allocation-in-memory.md

```diff
@@ -0,0 +1,84 @@
+---
+status: accepted
+date: 2024-05-21
+deciders: Kamil Kołodziej, Jakub Pisarek, Radosław Szuma
+consulted: Michał Śledź
+informed: Fishjam Team
+---
+# Route incoming traffic within the cluster without a need to provide room_id
+
+## Context and Problem Statement
+
+In the current architecture after we create a room in cluster we return on which instance (Jellyfish host) it was created.
+This means that Jellyfish client needs to make request to only that specific instance.
+Which effectively leads to the situation where client may need to keep mapping between room_id and jellyfish host.
+
+We would like to hide information about internal architecture of the cluster and the exact JF hosts as well as make it easier for clients so they can use only cluster endpoint.
+
+## Decision Drivers
+
+* Cluster architecture should be an internal information
+* `room_id` should be a sufficient information to route the requests
+
+## Considered Options
+
+* Store room allocation across the cluster using in-memory ETS
+* Store room allocation across the cluster using database
+* Leave it as it is and update the SKDs
+
+## Decision Outcome
+
+Chosen option: Store room allocation across the cluster using in-memory ETS.
+
+### Consequences
+
+- Routing within the cluster will take some of the responsibility from the client.
+- This responsibility is moved to the internal part of the Fishjam.
+- We may encounter some issues with data not being propagated quick enough or some message missing and will need to handle those scenarios.
+- We need to add some observability tools/metrics to allow us detect any issue within the cluster.
+- When it comes to the websocket we would prefer to forward it to the correct node instead of returning exact ws address or forward traffic.
```

Comment:
**question**: I have trouble understanding what exactly will be forwarded. Care to explain?

Reply:
Well we would like websocket to connect to the node where corresponding room is.
Since the request can be routed to any node we need to think how to handle that situation.

1. Based on the websocket RFC the redirect should be possible and client should comply - https://datatracker.ietf.org/doc/html/rfc6455

2. If that's not working we may need to return the exact link for websocket connection where we will return host for that given node - this is something what we want to avoid

3. Forward the data that is transmitted via websocket to the correct node and back (that's terrible I think).